### PR TITLE
[Merged by Bors] - chore(SetTheory/Ordinal/Basic): deprecate duplicate `Nat.cast` lemmas

### PR DESCRIPTION
**Pull request by vihdzp**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by urkud** [comment]
Otherwise LGTM. Thanks!
bors d+

File: Mathlib/SetTheory/Ordinal/Arithmetic.lean [hunk 1]
```diff
@@ -2099,6 +2099,10 @@ theorem Ordinal.not_bddAbove_compl_of_small (s : Set Ordinal.{u}) [hs : Small.{u
 
 namespace Ordinal
 
+instance instCharZero : CharZero Ordinal := by
+  refine ⟨fun a b h ↦ ?_⟩
+  rwa [← Cardinal.ord_nat, ← Cardinal.ord_nat, Cardinal.ord_inj, Nat.cast_inj] at h
```
[the remark this answers is not in the document]
Comment (vihdzp) [reply Mathlib/SetTheory/Ordinal/Arithmetic.lean]: I think this proof doesn't work? The build fails...

[the remark this answers is not in the document]
Comment (vihdzp) [reply Mathlib/SetTheory/Ordinal/Arithmetic.lean]: Yeah, the issue is that `Cardinal.ord_nat` isn't def-eq. And it probably shouldn't be either - "the natural `n` as an ordinal is the smallest order-type of a relation `Fin n → Fin n → Prop`" isn't a very nice definition.

**Pull request closed without merging**